source/tests/system/nidmm_driver_api_tests.cpp

```diff
@@ -0,0 +1,296 @@
+#include <gtest/gtest.h>
+
+#include "nidmm/nidmm_library.h"
+#include "nidmm/nidmm_service.h"
+
+namespace ni {
+namespace tests {
+namespace system {
+
+namespace dmm = nidmm_grpc;
+
+const int kDMMDriverApiSuccess = 0;
+
+class NiDMMDriverApiTest : public ::testing::Test {
+    protected:
+        NiDMMDriverApiTest()
+        {
+            ::grpc::ServerBuilder builder;
+            session_repository_ = std::make_unique<nidevice_grpc::SessionRepository>();
+            nidmm_library_ = std::make_unique<dmm::NiDMMLibrary>();
+            nidmm_service_ = std::make_unique<dmm::NiDMMService>(nidmm_library_.get(), session_repository_.get());
+            builder.RegisterService(nidmm_service_.get());
+
+            server_ = builder.BuildAndStart();
+            ResetStub();
+        }
+
+        virtual ~NiDMMDriverApiTest() {}
+
+        void SetUp() override
+        {
+            initialize_driver_session();
+        }
+
+        void TearDown() override
+        {
+            close_driver_session();
+        }
+
+        void ResetStub()
+        {
+            channel_ = server_->InProcessChannel(::grpc::ChannelArguments());
+            nidmm_stub_ = dmm::NiDMM::NewStub(channel_);
+        }
+
+        std::unique_ptr<dmm::NiDMM::Stub>& GetStub()
+        {
+            return nidmm_stub_;
+        }
+
+        int GetSessionId()
+        {
+            return driver_session_->id();
+        }
+
+        void expect_api_success(int error_status)
+        {
+            EXPECT_EQ(kDMMDriverApiSuccess, error_status) << get_error_message(error_status);
+        }
+
+        void initialize_driver_session()
+        {
+            ::grpc::ClientContext context;
+            dmm::InitWithOptionsRequest request;
+            request.set_resource_name("SimulatedDMM");
+            request.set_option_string("Simulate=1, DriverSetup=Model:4065; BoardType:PCI");
+            request.set_session_name("");
+            request.set_reset_device(false);
+            request.set_id_query(false);
+            dmm::InitWithOptionsResponse response;
+
+            ::grpc::Status status = GetStub()->InitWithOptions(&context, request, &response);
+            driver_session_ = std::make_unique<nidevice_grpc::Session>(response.vi());
+
+            ASSERT_TRUE(status.ok());
+            ASSERT_EQ(kDMMDriverApiSuccess, response.status());
```

Comment:
Is there difference between ASSERT_EQ and EXPECT_EQ (and ASSERT_TRUE/EXPECT_TRUE)?

`close_driver_session` below seems to use EXPECT vs you are using ASSERT here. If you can use EXPECT, then this line can be changed to `expect_api_success` and line above to use EXPECT_TRUE for consistency.

Reply:
changed it to use `expect_api_success` and EXPECT instead of ASSET